pyClient/README.md

```diff
@@ -45,11 +55,226 @@ and fails to fetch it (or fails to fetch the right version).
+# public address (zeth-address.json.pub)
+$ zeth gen-address
+
+# Share the public address with other users
+$ cp zeth-address.json.pub <destination>
+```
+
+With these files in place, `zeth` commands invoked from inside this directory
+can perform actions on behalf of Alice.  We call this Alice's *client directory*
+below, and assume that all commands are executed in a directory with these
+files.
+
+## Receiving transactions
+
+The following command scans the blockchain for any new transactions which
+generate Zeth notes indended for the public address `zeth-address.json.pub`:
+
+```console
+# Check all new blocks for notes addressed to `zeth-address.json.pub`,
+# storing them in the ./notes directory.
+(env)$ zeth sync
+```
+
+Any notes found are stored in the `./notes` directory, as files containing the
+secret data required to spend them.  This forms a very primitive Zeth "wallet".
+
+```console
+# List all notes received by this client
+$ zeth ls-notes
+```
+lists information about all known notes belonging to the current user.
+
+## Mix command
+
+The `zeth mix` command is used to interact with a deployed Zeth Mixer instance.
+The command accepts the following information:
+
+**Input Notes.** Zeth notes owned by the current client, which should be visible
+via `zeth ls-notes`.  Either the integer "address" or the truncated commitment
+value (8 hex chars) can be used to specify which notes to use as inputs.
+
+**Output Notes.** Given as pairs of Zeth public address and value, separated by
+a comma `,`. The form of the public address is exactly as in the
+`zeth-address.json.pub` file. That is, two 32 byte hex values separated by a
+colon `:`.
+
+**Public Input.** Ether or ERC20 token value to deposit in the mixer.
+
+**Public Output.** Ether or ERC20 tokens value to be withdrawn from the mixer.
+
+Some examples are given below
+
+### Depositing funds
+
+A simple deposit consists of some public input (ether or tokens), and the
+creation of Zeth notes.
+
+```console
+# Deposit 10 ether from `eth-address`, creating Zeth notes owned by Alice
+(env)$ zeth mix --out <public-zeth-address>,10 --vin 10
+```
+where `<public-zeth-address>` is the contents of `zeth-address.json.pub`.
+
+### Privately send a ZethNote to another user
+
+To privately transfer value within the mixer, no public input / output is
+required. Unspent notes (inputs) and destination addresses and output note
+values are specified.
+
+```console
+$ zeth ls-notes
+b1a2feaf: value=200, addr=0
+eafe5f84: value=100, addr=2
+
+$ zeth mix \
+    --in eafe5f84 \                       # "eafe5f84: value=100, addr=2"
+    --in 0 \                              # "b1a2feaf: value=200, addr=0"
+    --out d77f...0e00:cc7c....7f76,120 \  # 120 to this addr
+    --out 3a43...fd3b:9fc8....b838,180    # 180 to this addr
+```
+
+### Withdrawing funds from the mixer
+
+Specify the note(s) to be withdrawn, and the total value as public output:
+```console
+$ zeth mix --in eafe5f84 --vout 100
```

Comment:
Does the command crash if `--vout 100` is =/= from the value of the given note?
Say, what if I run:
```
zeth mix --in b1a2feaf --vout 100
```

Does it automatically create another note a value 100 belinging to me, and withdraws 100, or does it crash?

Reply:
Quickly checked. This would crash: https://github.com/clearmatics/zeth/blob/pyclient-cli/pyClient/commands/zeth_mix.py#L60-L61

Reply:
Hopefully, it outputs an error message :) but yes, it fails upfront.
We could add some support for change etc, btu for now I just enforce that everything is explicit and the values balance correctly.